### PR TITLE
Add helper command line arg to override active_work

### PR DESCRIPTION
**Pull request by andyleiserson**

### CODE REVIEW

File: pre-commit [hunk 1]
```diff
@@ -102,6 +102,9 @@ check "Tests" \
 check "Web tests" \
     cargo test -p ipa-core --no-default-features --features "cli web-app real-world-infra test-fixture compact-gate"
 
+check "Web tests (descriptive gate)" \
+    cargo test -p ipa-core --no-default-features --features "cli web-app real-world-infra test-fixture"
+
```
Comment (eriktaubeneck) [inline pre-commit:107]: maybe a dumb question, but how is descriptive gate related to the active work?

**Review by eriktaubeneck — APPROVED** [review]
one question, but otherwise looks good. thanks for adding this.

the missing coverage seems to be small changes in areas already missing coverage.

File: ipa-core/src/helpers/gateway/mod.rs [hunk 2]
```diff
@@ -98,6 +102,7 @@ impl Gateway {
         mpc_transport: MpcTransportImpl,
         shard_transport: ShardTransportImpl,
     ) -> Self {
+        tracing::info!("active_work = {}", config.active);
```
Comment (akoshelev) [inline ipa-core/src/helpers/gateway/mod.rs:105]: maybe `debug`?

**Review by akoshelev — APPROVED** [review]

File: pre-commit (hunk 1 above)
Comment (andyleiserson) [reply pre-commit:107]: It's not, but some of the plumbing I had to change to add the option broke a test case that only runs with descriptive gate, that is run by the coverage job, but not by `pre-commit`, so I added this.

File: ipa-core/src/helpers/gateway/mod.rs (hunk 2 above)
Comment (andyleiserson) [reply ipa-core/src/helpers/gateway/mod.rs:105]: The reason I made it `info` is that there is logic in `helpers::transport::query` that configures the active work dynamically to be no more than the query input size. At the time I didn't see the note in the commit description "Input size constraint comes from our CI that is so slow, it cannot allow very large active work sizes.", and the logic didn't make sense to me -- but I figured that if we really do need to be tuning it adaptively per query, then we probably also want to log it per query, at a level that we don't need to restart the server in debug mode to see.

But now that I have seen the commit message, it makes more sense. I will add the comment about CI in the code though.

**Pull request merged**